### PR TITLE
fix(docker): Add edge tag to Docker images, document when latest tag will appear

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

`latest` is the default Docker tag, but it's missing from published Zebra images:
https://hub.docker.com/r/zfnd/zebra/tags

This makes typical Docker commands fail:
```sh
$ docker run zfnd/zebra
Unable to find image 'zfnd/zebra:latest' locally
docker: Error response from daemon: manifest for zfnd/zebra:latest not found: manifest unknown: manifest unknown.
See 'docker run --help'.
```

### Specifications

https://github.com/docker/metadata-action#latest-tag

## Solution

- Add `edge` tag to published Docker images
- Document when latest tag will appear

## Review

Anyone can review this PR.

It is only a release blocker if we want users to be able to easily use Docker images.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

**Review by gustavovalverde — CHANGES_REQUESTED** [review]
I wouldn't add a `latest` tag to non-stable releases.

When a `<pre-release>` information is included in the image tag, the `latest` tag won't be automatically added as this one is, by default, reserved for stable images.

For example, we will release `1.0.0` and this will also auto-add the `latest` tag, but if we then publish `1.0.1-alpha.1`, we wouldn't want users to download an alpha version if they add the `latest` tag, but instead keep downloading `1.0.0` until `1.0.1` is published

This is explained with more detail here: https://github.com/ZcashFoundation/zebra/pull/5138#discussion_r971425816

**Comment by teor2345** [comment]
> I wouldn't add a `latest` tag to non-stable releases.
> 
> When a `<pre-release>` information is included in the image tag, the `latest` tag won't be automatically added as this one is, by default, reserved for stable images.
> 
> For example, we will release `1.0.0` and this will also auto-add the `latest` tag, but if we then publish `1.0.1-alpha.1`, we wouldn't want users to download an alpha version if they add the `latest` tag, but instead keep downloading `1.0.0` until `1.0.1` is published
> 
> This is explained with more detail here: [#5138 (comment)](https://github.com/ZcashFoundation/zebra/pull/5138#discussion_r971425816)

Thanks, I summarised that explanation next to the config that will create the `latest` tag.

**Review by gustavovalverde — DISMISSED** [review]
LGTM

**Comment by teor2345** [comment]
Some jobs didn't run on this PR, but the workflow rules contain `.github/workflows/build-docker-image.yml`.

I'll try re-running it, maybe GitHub is having issues.

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
Jobs seem to be totally unresponsive to branch pushes as well:
2662F1CE-5393-445D-B5CA-8D9DB64549EC

**Comment by teor2345** [comment]
Ok, closing and re-opening seemed to work.

File: .github/workflows/build-docker-image.yml [hunk 1]
```diff
@@ -71,16 +74,21 @@ jobs:
...
             type=semver,pattern={{version}}
             type=semver,pattern={{major}}.{{minor}}
             type=semver,pattern={{major}}
+            type=ref,event=tag
             type=ref,event=branch
             type=ref,event=pr
             type=sha
+            # edge is the latest commit on the default branch.
+            # We only have edge builds for `us-docker.pkg.dev`, because DockerHub doesn't keep up with every `main` branch commit.
+            type=edge,enable=${{ !env.USE_DOCKERHUB && is_default_branch }}
```
Comment (teor2345) [inline .github/workflows/build-docker-image.yml]:
```suggestion
            type=edge,enable=${{ !env.USE_DOCKERHUB && {{is_default_branch}} }}
```

**Comment by teor2345** [comment]
Looks like the workflow syntax was wrong, but the error only showed up on the workflow details screen, because the mistake was in a sub-workflow.

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
Ok, that should all be fixed now, some of my changes didn't work.

**Review by gustavovalverde — APPROVED** [review]

**Comment by gustavovalverde** [comment]
@Mergifyio update

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
It looks like some of the instances for this PR got deleted about 10 hours ago, before the tests had finished. I'm not sure why, the instance delete job wasn't running.
> Error: google-github-actions/ssh-compute failed with: failed to execute gcloud command `gcloud compute ssh lwd-grpc-wallet-5312-merge-9e47110 --zone us-central1-a ...
> - The resource 'projects/zealous-zebra/zones/us-central1-a/instances/lwd-grpc-wallet-5312-merge-9e47110' was not found

https://github.com/ZcashFoundation/zebra/actions/runs/3189762463/jobs/5209366519#step:6:99
https://github.com/ZcashFoundation/zebra/actions/runs/3189762463/jobs/5209407086#step:6:99

**Pull request merged**